Dec2022Demo/test_S1.jl

```diff
@@ -0,0 +1,116 @@
+types′ = LabelledPetriNet([:Pop],
+    :infect=>((:Pop, :Pop)=>(:Pop, :Pop)),
+    :disease=>(:Pop=>:Pop),
+    :strata=>(:Pop=>:Pop))
+types = map(types′, Name=name->nothing)
+
+
+SIRD = LabelledPetriNet([:S, :I, :R, :D],
+    :inf => ((:S,:I) => (:I,:I)),
+    :recover => (:I=>:R),
+    :death => (:I=>:D));SIRD = LabelledPetriNet([:S, :I, :R, :D],
+    :inf => ((:S,:I) => (:I,:I)),
+    :recover => (:I=>:R),
+    :death => (:I=>:D));
+AlgebraicPetri.Graph(SIRD)
+
+SIRD_typed = homomorphism(SIRD, types;
+initial=(T=[1,2,2],I=[1,2,3,3],O=[1,2,3,3]),
+type_components=(Name=x->nothing,))
+@assert is_natural(SIRD_typed)
+
+
+p_i(u,t) = 0.000025-0.000025*cos(t/100*2*pi)
+# sigmoid(x) = 1/(1-exp(-x))
+# p_i(u,t) = 0.00005*(1/(1+exp(-(t-30))))
+p_r(u,t) = 0.005+0.005*cos(t/100*4*pi+pi/4)
+p_d(u,t) = 0.001+0.001*cos(t/100*8*pi+pi/2)
+
+p_t = [p_i,p_r,p_d]
+
+tv_rxn = vectorfield(SIRD)
+tspan = (0,1000)
+u0 = [999,1,0,0]
+tv_prob = ODEProblem(tv_rxn, u0, tspan, p_t)
+tv_sol = solve(tv_prob, Tsit5())
+plot(tv_sol)
+
+p_i(u,t) = 0.000025-0.000025*cos(t/100*2*pi)
+p_r(u,t) = 0.01
```

Comment:
If this were a Pluto notebook, these could be sliders, see: https://github.com/AlgebraicJulia/ASKEM.jl/blob/32099bb026625a24ca061cde7da22ee5ac754879/notebooks/ceiling.jl#L60.